src/pe/debug.rs

```diff
@@ -131,60 +326,127 @@ pub struct CodeviewPDB70DebugInfo<'a> {
+                return Err(error::Error::Malformed(format!(
+                    "ImageDebugDirectory size of data seems wrong: {:?}",
+                    idd.size_of_data
+                )));
+            }
+            let filename_length = filename_length as usize;
 
-        // check the codeview signature
-        let codeview_signature: u32 = bytes.gread_with(&mut offset, scroll::LE)?;
-        if codeview_signature != CODEVIEW_PDB70_MAGIC {
-            return Ok(None);
+            // check the codeview signature
+            let codeview_signature: u32 = bytes.gread_with(&mut offset, scroll::LE)?;
+            if codeview_signature != CODEVIEW_PDB70_MAGIC {
+                return Ok(None);
+            }
+
+            // read the rest
+            let mut signature: [u8; 16] = [0; 16];
+            signature.copy_from_slice(bytes.gread_with(&mut offset, 16)?);
+            let age: u32 = bytes.gread_with(&mut offset, scroll::LE)?;
+            if let Some(filename) = bytes.get(offset..offset + filename_length) {
+                Ok(Some(CodeviewPDB70DebugInfo {
+                    codeview_signature,
+                    signature,
+                    age,
+                    filename,
+                }))
+            } else {
+                Err(error::Error::Malformed(format!(
+                    "ImageDebugDirectory seems corrupted: {:?}",
+                    idd
+                )))
+            }
+        } else {
+            // CodeView debug info not found
+            Ok(None)
         }
+    }
+}
+
+/// Represents the `IMAGE_DEBUG_VC_FEATURE_ENTRY` structure
+#[repr(C)]
+#[derive(Debug, PartialEq, Copy, Clone, Default)]
+pub struct VCFeatureInfo {
+    /// The count of pre-VC++
+    pub pre_vc_plusplus_count: u32,
+    /// The count of C and C++
+    pub c_and_cplusplus_count: u32,
+    /// The count of guard stack
+    pub guard_stack_count: u32,
+    /// The count of SDL
+    pub sdl_count: u32,
+    /// The count of guard
+    pub guard_count: u32,
+}
+
+impl<'a> VCFeatureInfo {
+    pub fn parse(
+        bytes: &'a [u8],
+        idd: ImageDebugDirectoryIterator<'_>,
+    ) -> error::Result<Option<Self>> {
+        Self::parse_with_opts(bytes, idd, &options::ParseOptions::default())
+    }
 
-        // read the rest
-        let mut signature: [u8; 16] = [0; 16];
-        signature.copy_from_slice(bytes.gread_with(&mut offset, 16)?);
-        let age: u32 = bytes.gread_with(&mut offset, scroll::LE)?;
-        if let Some(filename) = bytes.get(offset..offset + filename_length) {
-            Ok(Some(CodeviewPDB70DebugInfo {
-                codeview_signature,
-                signature,
-                age,
-                filename,
+    pub fn parse_with_opts(
+        bytes: &'a [u8],
+        idd: ImageDebugDirectoryIterator<'_>,
+        opts: &options::ParseOptions,
+    ) -> error::Result<Option<Self>> {
+        let idd = idd.collect::<Result<Vec<_>, _>>()?;
```

Comment:
do you really need to allocate the vec here? even iterating twice is probably better